### PR TITLE
Use fs API to load serialized PSL

### PR DESCRIPTION
**Pull request by mjethani**
The purpose of this PR is to switch to the filesystem API for loading the serialized public suffix list. The reason we need to do this is that we want to benchmark memory usage, and `require()` would keep the string in memory, which is not how it works in the browser extension.

### CODE REVIEW

File: platform/nodejs/index.js [hunk 1]
```diff
@@ -25,6 +25,12 @@
 
 import { createRequire } from 'module';
 
+import { readFileSync } from 'fs';
+import { dirname, resolve } from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
```
Comment (mjethani) [inline platform/nodejs/index.js:32]: [`__dirname`](https://nodejs.org/docs/latest/api/modules.html#modules_dirname) is a build-in variable in Node.js, but it's not available in ES modules. This is the common workaround. We want to know the directory in which this file exists.

File: platform/nodejs/index.js [hunk 2]
```diff
@@ -106,8 +116,13 @@ function pslInit(raw) {
 
         // Use serialized version if available
         try {
-            serialized = require('./build/publicsuffixlist.json');
+            // Use loadJSON() because require() would keep the string in memory.
+            serialized = loadJSON('build/publicsuffixlist.json');
         } catch (error) {
+            if (process.env.npm_lifecycle_event !== 'install') {
+                // This should never happen except during package installation.
+                console.error(error);
```
Comment (mjethani) [inline platform/nodejs/index.js:124]: This should help just in case the build process breaks.

**Comment by gorhill** [comment]
On the other hand, I suppose the new imports will contribute to increase the memory usage?

**Comment by mjethani** [comment]
> On the other hand, I suppose the new imports will contribute to increase the memory usage?

Hmm … that's a good point. I'm actually not sure. Let me test it out.

My expectation is that Node.js already loads that code one way or another (e.g. via `require()` itself). Meanwhile the JSON file is 235 KiB.

**Comment by mjethani** [comment]
> Anyway, `run.js` already imports `fs` and `path`, so not something to worry about.

For what it's worth, the way we'll do this is that we'll take an initial snapshot, then load the engine, then take another snapshot and compare the difference. If these imports add to the memory footprint by themselves then we can always import them first in `run.js`.

This is already the case for the fs and path modules, as you point out, but if there are any other ones we can deal with it by loading them first.

**Pull request merged**